### ISSUE TITLE
Rethink exception rethrowing strategy

### ISSUE THREAD

**Issue by ezyang** [pytorch/torchdynamo#1754]
### 🐛 Describe the bug

After https://github.com/pytorch/pytorch/pull/87440 there are a few things that no longer make sense in the new regime that we should think about changing.

* @mlazos remarked to me that now that we aren't suppressing errors by default, we shouldn't log them to stderr. This makes sense. Essentially, IF we suppress an error, we should log it to stderr; otherwise, we shouldn't.
* Rethrowing exceptions into TorchDynamoInternalError or similar error classes is annoying, because it means by default you get an uninformative stack trace at the bottom (and have to scroll up to find the useful stack trace). We should consider NOT rethrowing. The rethrowing is also why benchmark fails to report accurate exception info.
* It would be useful to know contexts where minification can occur, and if an exception is thrown in such a context, add some extra context saying that "hey, you can run the minifier like this to get a minified repro". This is in tension with bullet two.

cc @Chillee @anijain2305 @jansel

### Error logs

n/a

### Minified repro

n/a

**Comment by mlazos** [comment]
Just found https://docs.python.org/3/library/traceback.html#:~:text=If%20chain%20is%20true%20(the%20default)%2C%20then%20chained%20exceptions%20(the%20__cause__%20or%20__context__%20attributes%20of%20the%20exception)%20will%20be%20printed%20as%20well%2C%20like%20the%20interpreter%20itself%20does%20when%20printing%20an%20unhandled%20exception.

We actually can just remove the chained exceptions from the trace quite easily. I'll do this for non-verbose mode

**Comment by ezyang** [comment]
FWIW, as a developer, I DON'T want to lose the inner exception; that's the exception that tells me where the bug in dynamo is. The rethrown exception trace is useless for me.

**Comment by mlazos** [comment]
Yeah I'm dumb, I thought it would only print the inner one. Late night hope.
